news/_posts/2020-02-11-phpunit-security-issue-post-analysis.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: post
+title:  "Security issue with PHPUnit, post-incident analysis"
+subtitle: "The mysterious development dependencies!"
+date:   2020-02-11 08:00:00
+authors: [ PierreRambaud, PrestaShop ]
+icon: icon-chat
+image: /assets/images/theme/meta-logo-build.png
+tags: [maintenance, patch, modules]
+---
+
+Early January, we encountered a [security issue with PHPUnit](https://build.prestashop.com/news/critical-security-vulnerability-in-prestashop-modules/) in some modules that allowed attackers to perform arbitrary code execution without authorization through the PHPUnit dependency. This vulnerability was discovered through a merchant whose shops were compromised. 
+
+## Background
+
+At PrestaShop, there is an automatic system on the Addons Marketplace which gets the module from the [PrestaShop-modules](https://github.com/PrestaShop/PrestaShop-modules) repository and uses Composer to install dependencies to be distributed by the API. During this step, by mistake, development dependencies have been built inside.
+ 
+This was a code error, fixed after a short time, but the bad archives were already downloaded and unfortunately some archives which contained PHPUnit were not properly cleaned up.
+
+Moreover, when the PrestaShop software upgrades a module, it only overrides already-existing files, so files that were inserted once were not removed, even if a more recent version of dependency was provided. 
+
+This means that if a module was installed once with unprotected files, these files would remain there until someone removes or updates them. This problem also applies to themes.
+
+## Communication
+
+The security team reached out to other major CMS teams in order to inform them about the issue so they could check whether the issue affected modules or plugins on their side.
+ 
+Since last year's [Google CMS Security summit](https://build.prestashop.com/news/we-were-at-the-cms-security-summit-with-google/), we have been part of a Slack channel that gathers many people from the CMS ecosystem (Wordpress, Joomla, host providers, etc.) to share security issues. This allowed us to share frequent information with other CMS people.
```

Comment:
```suggestion
Since last year's [Google CMS Security summit](https://build.prestashop.com/news/we-were-at-the-cms-security-summit-with-google/), we have been part of a Slack channel that gathers many people from the CMS ecosystem (Wordpress, Joomla, host providers, etc.) to share security issues. This allows us to share information frequently.
```